doc/api/policy.md

```diff
@@ -184,4 +193,108 @@ module.exports = function fn(...args) {
+
+This feature is experimental and needs to be enabled by passing either or both
+of the `--policy-deny=` or `--policy-grant` CLI flags to Node.js.
+
+Both flags take a comma-separated list of permission identifiers as described
+below. For example, `--policy-deny=fs,net.udp`.
+
+### Permissions
+
+The permission identifiers are:
+
+* `*` - Wildcard for all permissions.
+* `special` - Special categories are denied by default if any other permission
+  is denied.
+  * `special.inspector` - Controls ability to use Inspector APIs (not yet
+    active)
+  * `special.addons` - Controls ability to load native addons / use
+    `process.dlopen()`
+  * `special.child_process` - Controls ability to spawn child processes
+* `workers` - Controls ability to spawn worker threads
+* `fs` - Controls all file system operations
+  * `fs.in` - Controls file system read operations
+  * `fs.out` - Controls file system write operations
+* `user` - Controls access to user data methods (e.g. `os.homedir()`,
+  `os.userInfo()`)
+* `net` - Controls access to networking APIs
+  * `net.udp` - Controls access to connectionless UDP
+  * `net.dns` - Controls access to DNS APIs.
+  * `net.tcp` - Controls access to all TCP APIs
+    * `net.tcp.in` - Controls access to TCP `listen()` operations
+    * `net.tcp.out` - Controls access to TCP `connect()` operations
+  * `net.tls` - Controls access to TLS wrap APIs.
+    * `net.tls.log` - Controls access to TLS keylogging
+* `process` - Controls access to APIs that change process state
+* `timing` - Controls access to timing APIs (e.g. `process.hrtime()`,
+  `performance.now()`, `performance.mark()`, etc. Has no impact on the
+  use of `new Date()`)
+* `signal` - Controls ability to send signals (e.g. `process.kill()`)
+* `experimental` - Controls access to experimental APIs
+  * `experimental.wasi` - Controls access to experimental WASI apis.
+
+Permission identifiers are hierarchical. For instance
+`--policy-deny=net` will deny access to all `net.*` identifiers.
+
+### `--policy-deny` and `--policy-grant`
+
+The `--policy-deny` and `--policy-grant` command-line flags specify the
+policy permissions for the Node.js process at startup. Upon initialization,
+the `--policy-deny` list will be processed first, followed by the
+`--policy-grant` list.
+
+For instance,
+
+```
+--policy-deny=* --policy-grant=fs.in
+```
+
+Will deny all permissions except for `fs.in`, while:
+
+```
+--policy-deny=fs.in --policy-grant=*
+```
+
+Will result in all permissions being granted.
+
+To disable all network capabilities other than DNS, the combination would be:
+
+```
+--policy-deny=net --policy-grant=net.dns
+```
+
+### `process.policy` API
+
+The `process.policy` API allows the current enforced policy to be
+inspected at runtime and permits the ability to further restrict the
+current policy via the `process.policy.deny()` method. It is not
+possible to grant permissions via the JavaScript API.
+
+#### `process.policy.granted(permission)`
```

Comment:
Wdyt if there is no such method and we add a flag/conf where any policy denial stops the current process with an error?

A script can always `try...catch` around a call that might fail due to a policy denial. But in the case of a untrusted script, I'd prefer to have a way to prevent it from running if it tries to do anything I don't want. 

Alternatively, the call to `granted` could kill the process too in this "high-secure" mode

Reply:
This actually leads me to another question which is enabling reporting of violations.

@vdeturckheim Would providing a `'policyViolation'` event suffice to handle exiting?

Reply:
@bengl I'd be more comfortable with a hard coded exit. My attack scenario here would be something like a malicious module used in different contexts that check permission to act or not.
For instance, a linting module usually would only run in a process with `fs.in` right, but if a feature allows it to write the report on the disk or to a remote service, the module could act maliciously "when in the right situation".
Having a flag that stop the process without any possibility to intercept when a denial/granting check happens would actually prevent allow to identify such module.

Reply:
Yeah on second thought, reporting can be handled more safely by whatever process monitoring tool is used.


+1 on optionally exiting on violation.

Reply:
Such flag could also be added later btw.